config/intellij-idea-inspection-scope.xml

```diff
@@ -2,5 +2,5 @@
      all excludes should be specified in build configuration -->
 <component name="DependencyValidationManager">
   <scope name="Checkstyle Inspection Scope"
-      pattern="!file[checkstyle]:target//*&amp;&amp;!file:src/test/resources*//**&amp;&amp;!file:src/it/resources*//**&amp;&amp;!file:src/site/resources/js/google-analytics.js&amp;&amp;!file:src/site/resources/styleguides*//**&amp;&amp;!file:config/intellij-idea-inspections.properties&amp;&amp;!file[checkstyle]:.ci-temp//*&amp;&amp;!file[checkstyle]:bin//*&amp;&amp;!file[checkstyle]:.settings//*&amp;&amp;!file:.classpath&amp;&amp;!file:.project&amp;&amp;!file:.circleci/config.yml&amp;&amp;!file:config/projects-to-test/openjdk-projects-to-test-on.config&amp;&amp;!file:config/projects-for-no-exception-javadoc.config&amp;&amp;!file:.ci/pitest-survival-check-xml.groovy"/>
+      pattern="!file[checkstyle]:target//*&amp;&amp;!file:src/test/resources*//**&amp;&amp;!file:src/it/resources*//**&amp;&amp;!file:src/site/resources/js/google-analytics.js&amp;&amp;!file:src/site/resources/styleguides*//**&amp;&amp;!file:config/intellij-idea-inspections.properties&amp;&amp;!file[checkstyle]:.ci-temp//*&amp;&amp;!file[checkstyle]:bin//*&amp;&amp;!file[checkstyle]:.settings//*&amp;&amp;!file:.classpath&amp;&amp;!file:.project&amp;&amp;!file:.circleci/config.yml&amp;&amp;!file:config/projects-to-test/openjdk-projects-to-test-on.config&amp;&amp;!file:config/projects-for-no-exception-javadoc.config&amp;&amp;!file:.ci/pitest-survival-check-xml.groovy&amp;&amp;!file:target/generated-sources/antlr/com/puppycrawl/tools/checkstyle/grammar/java*//**"/>
```

Comment:
@romani , @nrmancuso ,
Is what I did here wrong?
the multisegment wildcard that I added is theoretically supposed to exclude all files and sub-dir of the pathspec.
But the violations in inspections continue to live on, meaning scope hasn't reduced..

Reply:
```
✔ ~/java/github/romani/checkstyle/target/generated-sources/antlr [master] 
$ ls -1
com
JavadocLexer.tokens
JavadocParser.tokens
JavaLanguageLexer.tokens
JavaLanguageParser.tokens
```
we already have `pattern="!file[checkstyle]:target//*` at the begining.
strange that it is not enough.
try 
`&amp;&amp;!file:**/generated-sources/antlr/com/puppycrawl/tools/checkstyle/grammar/**`

Reply:
do you have the same problem on local ?

Reply:
> we already have pattern="!file[checkstyle]:target//* at the begining.
strange that it is not enough.

Yes we do. I initially thought scope wasn't being restricted at /antlr because of the single-segment wildcard (only one level ls) that we used in our path pattern syntax. I already tried changing it to multi-segment (all levels below) but it was of no use. This is why I added precise pathspec which you see in the diff.

> try (this is what IDEA generated when I excluded this folder)
> `&amp;&amp;*!file[checkstyle]:target/generated-sources/antlr//*`

@romani , I'm still not sure if this will reach files of level *//java* because the pattern is still single-segment
https://cloud.google.com/eventarc/docs/path-patterns#path_pattern_syntax

Reply:
> do you have the same problem on local ?

IntelliJ is known to be smart sometimes.. but this seems on an other level 😅 
My tree looks fine on local with this scope pattern : 

![image](https://github.com/checkstyle/checkstyle/assets/148562451/1eee4aa0-8d69-49b9-90c7-6cd84b306091)

> there is new IDEA release, there is somthing about inspections https://youtrack.jetbrains.com/articles/IDEA-A-2100661888/IntelliJ-IDEA-2023.3.5-233.14808.21-build-Release-Notes

I couldn't find anything in their issue/pull that could explain what's happening here :(
That issue fixed in their release looks internal of IDEA, something related to it's error tab showing outdated errors
(shouldn't really bother us i think)

Reply:
@romani , As a completely different thought, could you please explain why we can't change the running/execution directory of inspections engine ? (https://github.com/checkstyle/checkstyle/pull/14604#issuecomment-1993271606)

I think doing this will make the inspection violations go away for checkstyle codebase in our ci-run-inspections

Reply:
We need scopes, as we need to get same violations by running of same version of Idea on locals. Shell file is is just a convenience for CI, not people.

Reply:
May be we just need to report issue on Idea , tha scopes are not working anymore in new versions.

Reply:
> May be we just need to report issue on Idea , tha scopes are not working anymore in new versions.

Sure.. I will follow-up with them.
@romani , could you please tell me what version of Idea you have been using ?

> We need scopes, as we need to get same violations by running of same version of Idea on locals. Shell file is is just a convenience for CI, not people.

Oh okay, Is there any other workaround for us until then ?

Reply:
> could you please tell me what version of Idea you have been using ?

I used old version, that was a exact version that out CI is using in master.
I downloaded new version, same as you have in this PR. I always matching what version in CI to be consistent.

> Is there any other workaround for us until then ?

we need to make IDEA to work on local with scopes, we might need to try few versions, even 2024.X, We need any version where scopes works.
Before issue we can try create scope file from scratch in IDEA UI.
If scopes are not working any where, we need to report issue to Jet Brains.

Reply:
> we need to make IDEA to work on local with scopes, we might need to try few versions, even 2024.X, We need any version where scopes works.

@romani , I have tried a few releases and `v2023.1.2` seems fine with scopes : 

![image](https://github.com/checkstyle/checkstyle/assets/148562451/364fdd3d-4294-4f72-8899-e024fbb91a60)

I know this isn't close to latest release, but should we try this release and maybe it will resolve our primary issue with yaml violation ?

> Before issue we can try create scope file from scratch in IDEA UI.

My tree produces the same problem as yours at https://github.com/checkstyle/checkstyle/pull/14604#discussion_r1527524704 by removing whole pattern and using only `!file[checkstyle]:target/generated-sources/antlr//*` also. 
Please bear with me and confirm this on your local too if possible, so I can proceed with collaborating with them

Reply:
>I know this release 2023.1.2 isn't close to latest release, but should we try this release and maybe it will resolve our primary issue with yaml violation ?

Yes, lets update up to version that still supports scopes.
Let's report scope problems to jetbrain.

Reply:
Have we tried to completely delete and regenerate (by UI) the scope file?

Reply:
> Have we tried to completely delete and regenerate (by UI) the scope file?

Yes @nrmancuso , I have completely removed the scope file from idea scopes and tried generating path tree by adding the pattern in the textField. It was still the same as https://github.com/checkstyle/checkstyle/pull/14604#discussion_r1528481172

Reply:
@romani , @nrmancuso , can we temporarily block this PR while #14696 is being worked on

Reply:
please report issue to JetBrains.
Ideally old scope files should work (but not critical), main point is that scopes should work if they are setup by Idea UI.

Reply:
@romani , Issue reported to jetBrains tracker at [IDEA-351113](https://youtrack.jetbrains.com/issue/IDEA-351113)